core/src/main/scala/za/co/absa/spline/harvester/dispatcher/HttpLineageDispatcher.scala

```diff
@@ -18,82 +18,93 @@ package za.co.absa.spline.harvester.dispatcher
 import za.co.absa.spline.harvester.dispatcher.httpdispatcher.rest.{RestClient, RestEndpoint}
-import za.co.absa.spline.harvester.dispatcher.httpdispatcher.{ProducerApiCompatibilityManager, ProducerApiVersion}
 import za.co.absa.spline.harvester.exception.SplineInitializationException
 import za.co.absa.spline.harvester.json.HarvesterJsonSerDe.impl._
 import za.co.absa.spline.producer.model.v1_1.{ExecutionEvent, ExecutionPlan}
 
-import scala.concurrent.duration._
+import javax.ws.rs.core.MediaType
 import scala.util.Try
 import scala.util.control.NonFatal
 
-object HttpLineageDispatcher extends Logging {
-  private val ProducerUrlProperty = "spline.producer.url"
-  private val ConnectionTimeoutMsKey = "spline.timeout.connection"
-  private val ReadTimeoutMsKey = "spline.timeout.read"
+/**
+ * HttpLineageDispatcher is responsible for sending the lineage data to spline gateway through producer API
+ */
+class HttpLineageDispatcher(restClient: RestClient)
+  extends LineageDispatcher
+    with Logging {
 
-  private val DefaultConnectionTimeout: Duration = 1.second
-  private val DefaultReadTimeout: Duration = 20.second
+  import za.co.absa.spline.harvester.dispatcher.HttpLineageDispatcher._
 
-  private object RESTResource {
-    val ExecutionPlans = "execution-plans"
-    val ExecutionEvents = "execution-events"
-    val Status = "status"
-  }
+  def this(configuration: Configuration) = this(HttpLineageDispatcher.createDefaultRestClient(configuration))
+
+  private val executionPlansEndpoint = restClient.endpoint(RESTResource.ExecutionPlans)
+  private val executionEventsEndpoint = restClient.endpoint(RESTResource.ExecutionEvents)
 
-  private object SplineHttpHeaders {
-    private val Prefix = "ABSA-Spline"
+  private val serverHeaders: Map[String, IndexedSeq[String]] = getServerHeaders(restClient)
+  private val apiVersion: Version = resolveApiVersion(serverHeaders)
+  private val modelMapper = ModelMapper.forApiVersion(apiVersion)
 
-    val ApiVersion = s"$Prefix-API-Version"
-    val ApiLTSVersion = s"$Prefix-API-LTS-Version"
-    val AcceptRequestEncoding = s"$Prefix-Accept-Request-Encoding"
+  private val requestCompressionSupported: Boolean =
+    serverHeaders(SplineHttpHeaders.AcceptRequestEncoding)
+      .exists(_.toLowerCase == Encoding.GZIP)
+
+  override def send(executionPlan: ExecutionPlan): String = {
+    val execPlanDTO = modelMapper.toDTO(executionPlan)
+    sendJson(execPlanDTO.toJson, executionPlansEndpoint)
   }
 
-  private def createHttpErrorMessage(msg: String, code: Int, body: String): String =
-    s"$msg. HTTP Response: $code $body"
+  override def send(event: ExecutionEvent): Unit = {
+    val eventDTO = modelMapper.toDTO(event)
+    sendJson(Seq(eventDTO).toJson, executionEventsEndpoint)
+  }
+
+  private def sendJson(json: String, endpoint: RestEndpoint) = {
+    val url = endpoint.request.url
+    logTrace(s"sendJson $url : $json")
```

Comment:
Since we are logging this, is there an easy way to format the json before it is logged?

Reply:
parse and re-render with pretty-print probably. Will do

Reply:
updated